### PR TITLE
Updated Binaries December 2023

### PR DESCRIPTION
**Pull request by martindevans**
- build run: https://github.com/SciSharp/LLamaSharp/actions/runs/7196891440
 - commit: https://github.com/ggerganov/llama.cpp/blob/9fb13f95840c722ad419f390dc8a9c86080a3700

This build includes AVX and AVX2 for MacOS x86_64. If anyone has an Intel Mac testing would be much appreciated!

Todo:
 - [x] Json serialization for `MetadataOverride`

Testing:
 - [x] Windows
 - [x] Linux
 - [ ] MacOS (ARM)
 - [x] MacOS (x86_64)

### CODE REVIEW

File: LLama/Native/LLamaKvCacheView.cs [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    /// <returns></returns>
+    [DllImport(libraryName, CallingConvention = CallingConvention.Cdecl)]
+    public static extern LLamaKvCacheView llama_kv_cache_view_init(SafeLLamaContextHandle ctx, int n_max_seq);
+
+    /// <summary>
+    /// Free a KV cache view. (use only for debugging purposes)
+    /// </summary>
+    /// <param name="view"></param>
+    [DllImport(libraryName, CallingConvention = CallingConvention.Cdecl)]
+    public static extern unsafe void llama_kv_cache_view_free(LLamaKvCacheView* view);
```
Comment (AsakusaRinne) [inline LLama/Native/LLamaKvCacheView.cs:71]: This is an amazing feature!

**Comment by martindevans** [comment]
Ok this is ready for review now. The CI has run tests for Windows, Linux and Mac (x64). Just MacOS ARM left!

**Review by AsakusaRinne — APPROVED** [review]
Great work!

**Comment by martindevans** [comment]
Some discussion about this in #350, which might end up being merged into this PR.

If you'd rather not wait, we can merge this PR and put out a release right away. The cudart stuff can be added in a patch update at any time.

**Comment by AsakusaRinne** [comment]
If it's convenient for you, then feel free to do so. :)

**Comment by SignalRT** [comment]
It's checked on OSX-ARM. It's Ok for me.

**Comment by martindevans** [comment]
Thanks for testing that @SignalRT.

I think I'll merge this one now, cudart stuff can come in a later PR.

**Pull request merged**